File: Week 1/mario/mario.c
#include <stdio.h>
#include "cs50.h"

int get_height(void);
void pyramid(int size);


int main(void)
{
    int size = get_height();

    pyramid(size);
}


int get_height(void)
{
    int height;

    do
    {
        height = get_int("Height: ");
    } 
    while ((height < 0) || (height > 8));
    
    printf("\n");

    return height;
}


void pyramid(int size)
{
    for (int i = 1; i <= size; i++)
    {
        int space = (size - i);
        
        for (int j = 0; j < space; j++)
        {
            printf(" ");
        }
        
        for (int n = 1; n <= i; n++)
        {
            printf("#");
        }
        
        printf("  ");
        
        for (int x = 1; x <= i; x++)
        {
            printf("#");
        }
    }   
}

File: Week 6/mario-more/mario.py
from cs50 import get_int

def main():

    while True:
        h = get_int("Height: ")

        if h > 8 or h < 1:
            continue
        else:
            break

    print_pyramid(h)


def print_pyramid(height):
    i = 1
    j = 1

    while i <= height:
        while j <= height:
            spaces = height - i
            print(" " * spaces, end='')
            print("#" * i, end='')
            print(" " * 2, end='')
            print("#" * i, end='')
            print()

            i += 1
            j += 1



if __name__ == "__main__":
    main()
